symengine/tests/basic/test_basic.cpp

```diff
@@ -314,40 +314,6 @@ TEST_CASE("Mul: Basic", "[basic]")
     REQUIRE(vec_basic_eq(r->get_args(), {}));
 }
 
-TEST_CASE("Diff: Basic", "[basic]")
-{
-    RCP<const Basic> r1, r2;
-    RCP<const Symbol> x  = symbol("x");
-    RCP<const Symbol> y  = symbol("y");
-    RCP<const Basic> i2  = integer(2);
-    RCP<const Basic> i3  = integer(3);
-    RCP<const Basic> i5  = integer(5);
-    RCP<const Basic> i10  = integer(10);
-    r1 = integer(5);
-    r2 = r1->diff(x);
-    REQUIRE(eq(*r2, *zero));
-
-    r1 = Rational::from_two_ints(*integer(2), *integer(3));
-    r2 = r1->diff(x);
```

Comment:
@Sumith1896 The purpose of this, was to remove member functions as stated in https://github.com/symengine/symengine/pull/527#issue-94633819 and https://github.com/symengine/symengine/pull/731#discussion_r48684668
Right now, I think the member function is being called, not the ones you wrote.
Call it as `diff(r1, x)` instead (and everywhere else) . And also remove `Basic::diff()` and `diff()` from all other classes.